llvm-general/src/LLVM/General/Internal/Module.hs

```diff
@@ -236,6 +236,19 @@ getDataLayout m = do
   dlString <- decodeM =<< FFI.getDataLayout m
   either fail return . runExcept . parseDataLayout A.BigEndian $ dlString
 
+-- | Build an LLVM.General.'Module' from a
+-- LLVM.General.AST.'LLVM.General.AST.Module' with a 'DataLayout'
+-- appropriate for the given 'TargetMachine'. This is important for
+-- native code generation as it impacts name mangling.
+withModuleFromASTForTargetMachine :: Context -> TargetMachine -> A.Module
```

Comment:
I'm not familiar with the part of LLVM that matches `llvm-general-pure`, so I don't know if something like this exists. It could well be a documentation fix, but I found that `withModuleFromAST` by itself was a bit of a trap because the output of `moduleLLVMAssembly` was then not usable on my machine.